extension/runtime/src/main/java/org/iris_events/producer/EventProducer.java

```diff
@@ -146,6 +165,60 @@ public void send(final Object message, final String userId, final boolean propag
         doSend(message, userId, propagate);
     }
 
+    public <T> T sendRpcRequest(final Object message, Class<T> responseType) {
+
+        final var requestId = UUID.randomUUID();
+        final var messageAnnotation = getMessageAnnotation(message);
+        final var eventName = ExchangeParser.getFromAnnotationClass(messageAnnotation);
+
+        final var replyToType = RpcResponseClassParser.getFromAnnotationClass(messageAnnotation).name();
+
+        final var replyTo = rpcMappingProvider.getReplyTo(replyToType);
+        if (replyTo == null) {
+            throw new IrisSendException("Can not send RPC request message with missing replyTo parameter.");
+        }
+
+        try (Channel channel = channelService.createChannel()) {
+            final var requestExchange = exchangeNameProvider.getRpcRequestExchangeName(eventName);
+            final var responseExchange = exchangeNameProvider.getRpcResponseExchangeName(replyTo);
+            final var requestQueueName = queueNameProvider.getRpcRequestQueueName(eventName);
+            final var responseQueue = queueNameProvider.getRpcResponseQueueName(replyTo);
+
+            channel.queueDeclare(responseQueue, false, false, true, null);
+            channel.queueBind(responseQueue, responseExchange, responseQueue);
+
+            final CompletableFuture<T> response = new CompletableFuture<>();
+            String ctag = channel.basicConsume(responseQueue, false, (consumerTag, delivery) -> {
+                final var messageObject = objectMapper.readValue(delivery.getBody(), responseType);
+                final var answerRequestId = delivery.getProperties().getHeaders().get(Headers.HEADER_REQUEST_ID).toString();
+
+                if (requestId.toString().equals(answerRequestId)) {
+                    response.complete(messageObject);
+                    channel.basicAck(delivery.getEnvelope().getDeliveryTag(), false);
+                } else {
+                    channel.basicNack(delivery.getEnvelope().getDeliveryTag(), false, false);
+                }
+            }, consumerTag -> {
+            });
+
+            final var headers = Map.of(Headers.HEADER_REQUEST_ID, (Object) requestId.toString());
```

Comment:
what about using basicProperties.messageId(...)?

Reply:
Will implement this solution and test.

Reply:
Done and tested.